docs/architecture/data-model.md

```diff
@@ -0,0 +1,105 @@
+layout: default
+title: Data Model
+parent: Architecture
+description: lakeFS Data Model explained
+nav_order: 2
+has_children: false
+---
+# Data Model
+{: .no_toc }
+
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+## Overview
+
+Since commits in lakeFS are immutable, they are easy to store on an immutable object store.
+
+Older commits are rarely accessed, while newer commits are accessed very frequently, a tiered storage approach can work very well - The object store is the source of truth, while local disk and even RAM can be used to cache the more frequently accessed ones.
+
+Since they are immutable - once cached, we only need to evict them when space is running out. There’s no complex invalidation that needs to happen.
+
+In terms of storage format, commits are be stored as [SSTables](https://en.wikipedia.org/wiki/Log-structured_merge-tree){: target="_blank" }, compatible with [RocksDB](https://rocksdb.org/){: target="_blank" }.
+
+SSTables were chosen as a storage format for 3 major reasons:
+
+1. Extremely high read throughput on modern hardware: using commits representing a 200m object repository (modeled after the S3 inventory of one of our design partners), we were able to achieve close to 500k random GetObject calls / second. This provides a very high throughput/cost ratio, probably as high as can be achieved on public clouds.
+1. Being a known storage format means it’s relatively easy to generate and consume. Storing it in the object store makes it accessible to data engineering tools for analysis and distributed computation, effectively reducing the silo effect of storing it in an operational database.
+1. The SSTable format supports [delta encoding for keys](https://github.com/facebook/rocksdb/wiki/PlainTable-Format#prefix-encoding){: target="_blank" } which makes them very space efficient for data lakes where many keys share the same common prefixes.
+
+Each lakeFS commit is represented as a set of contiguous, non-overlapping SSTables that make up the entire keyspace of a repository at that commit.
+
+## SSTable File Format ("Graveler File")
+
+lakeFS metadata is encoded into a format called **"Graveler"** - a standardized way to encode content-addressable key value pairs. This is what a Graveler file looks like:
+
+![Graveler File](../assets/img/graveler1.png)
+
+Each Key/Value pair (**"ValueRecord"**) is constructed of a `key`, an `identity` and a `value`.
+
+A simple identity could be, for example, a sha256 hash of the value’s bytes, but it could be any sequence of bytes that uniquely identifies the value. As far as Graveler is concerned, two `ValueRecord`s are considered identical, if their key and identity fields are equal.
+
+A Graveler file itself is content-addressable, i.e. similarly to Git, the name of the file is its identity.
+File identity is calculated based on the identity of the ValueRecords the file contains:
+
+
+<b>valueRecordID = h(h(valueRecord.key) + h(valueRecord.Identity))</b><br/>
```

Comment:
"+" is overloaded here (and, if this really were "+", "h(x+y)" would be no better than "x+y").  Consider using some other operator for ("structured") concatenation, perhaps `||`.  Particularly since this isn't really concatenation at almost any stage, we include lengths for software engineering reasons.